### PR TITLE
add paster command to add existing column types to the Data Dictionary

### LINKED ISSUE #84: Import existing column types into data dictionary?
We have a CKAN system that has been using the DataPusher, and we want to migrate to XLoader to improve performance and reduce administration overheads. However, we have thousands of resources parsed by messytables, and we want to preserve the type information (to format the data better for client applications), which XLoader doesn't by default.

To avoid updating each resource manually, we're looking at writing a paster command to inspect the existing columns and add Data Dictionary overrides for all of the non-text fields for when we move to XLoader.

Are there any obvious pitfalls with this approach?

### PR DESCRIPTION
**Pull request by ThrawnCA**
This resolves #84

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -99,6 +99,10 @@ Caveats
   or datestamp and re-load the file. There is scope to do this automatically in
   future.
 
+  * If you have existing data, you can pre-populate the Data Dictionary:
+
+     paster --plugin=ckanext-xloader migrate_types -c /etc/ckan/default/ckan.ini
```
Comment (davidread) [inline README.rst]: Would you mind moving this to the Upgrading from DataPusher section, say before step 2?

File: ckanext/xloader/cli.py [hunk 2]
```diff
@@ -230,3 +233,108 @@ def _print_status(self):
...
+            return
+
+        fields = []
+        for field in data_dict:
+            if field['type'] == 'text' and not self.options.include_text:
+                type_override = ''
+            else:
+                type_override = field['type']
+
+            if not 'info' in field:
```
Comment (davidread) [inline ckanext/xloader/cli.py]:
```suggestion
            if 'info' not in field:
```
and just some other minor flake8 errors to fix in this file, to make Travis pass

File: README.rst (hunk 1 above)
Comment (ThrawnCA) [reply README.rst]: Can do, probably on Monday.

File: ckanext/xloader/cli.py (hunk 2 above)
Comment (ThrawnCA) [reply ckanext/xloader/cli.py]: This is done.

**Comment by duttonw** [comment]
Looks good

**Comment by davidread** [comment]
Thanks both!

**Pull request merged**